specifications/transactions/offchain-api/README.md

```diff
@@ -0,0 +1,256 @@
+
+* VASP A knows VASP B’s on-chain account and obtains the url from VASP B's `DualAttestation::Credential::base_url` resource
+* VASP A connects to this endpoint and establishes a tls-encrypted HTTP channel
+* VASP A sends a request across the channel
+  * VASP A, first, constructs message containing a `CommandRequestObject`
+  * VASP A, then, signs the message it with the Ed25519 private key that maps to the public key stored at VASP A’s `DualAttestation::Credential::compliance_public_key`
+  * VASP A, finally transmits the signed message across the channel
+* VASP B receives the `CommandRequestObject`
+  * VASP B, first, validates the request message
+  * VASP B, then, evaluates and responds to the message constructing `CommandResponseObject`
+  * VASP B signs the response with the Ed25519 private key that maps to the public key stored at VASP B’s `DualAttestation::Credential::compliance_public_key`
+* VASP A receives the `CommandResponseObject`, validates the response, and concludes the exchange
+
+## Specification
+### On-Chain Account Setup
+
+A VASP’s on-chain account must be created and set up in accordance with the standard VASP account creation process.
+
+Of particular note for off-chain APIs is the compliance key and base url, [DualAttestation::Credential](https://github.com/diem/diem/blob/release-1.1/language/stdlib/modules/DualAttestation.move)’s `compliance_public_key` and `base_url`. During account setup, a VASP should set their compliance key and base url which will be stored under the Parent VASP account, see [DIP-2](https://github.com/diem/dip/blob/master/dips/dip-2.md) for more information on on-chain roles.
+
+The compliance key value is the Ed25519 key with which a VASP signs travel rule statements which are verified on-chain and all off-chain requests and responses.
+
+The `base_url` frameowrk does not support multiplexing and should not be shared across different providers and accounts.
+
+### HTTP End-point
+
+Each VASP exposes an HTTPS POST end point at `https://&lt;hostname>:&lt;port>/&lt;protocol_version>/command`. The protocol_version is v2 for the currente off-chain APIs.
+
+The base url value should be the url without path `/&lt;protocol_version>/command`, e.g. `https://&lt;hostname>:&lt;port>`.
+
+### HTTP Headers
+
+All HTTP requests must contain:
+
+* A header `X-REQUEST-ID` with a unique UUID (according to [RFC4122](https://tools.ietf.org/html/rfc4122) with “-”'s included) for the request, used for tracking requests and debugging. Responses must have the same string in the `X-REQUEST-ID` header value as the requests they correspond to.
+* A header `X-REQUEST-SENDER-ADDRESS` with the HTTP request sender’s VASP [DIP-5](https://dip.diem.com/dip-5/) address used in the command object.
```

Comment:
Specifically this needs to be a ParentVasp account or can it be a child?

Reply:
good call, but parent

Reply:
as it says "address used in the command object.", it can be child vasp.

Reply:
Parent VASP accounts are the only ones that have compliance key resources in their account so it MUST be a parent vasp account, so we just need to be much more explicit here.

Reply:
that's why we need a spec :)